docs/develop/financial-accounting.md

```diff
@@ -0,0 +1,187 @@
+---
+sidebar_position: 2
+---
+
+# Financial Accounting
+
+For developers with non-financial backgrounds, TigerBeetle's use of accounting concepts like debits
+and credits may be one of the trickier parts to understand. However, these concepts have been the
+language of business for hundreds of years, so we promise it's worth it!
+
+This page goes a bit deeper into debits and credits, double-entry bookkeeping, and how to think
+about your accounts as part of a type system.
+
+## Building Intuition with Two Simple Examples
+
+If you have an outstanding loan and owe a bank $100, is your balance $100 or -$100? Conversely, if
+you have $200 in your bank account, is the balance $200 or -$200?
+
+Thinking about these two examples, we can start to build an intuition that the **positive or
+negative sign of the balance depends on whose perspective we're looking from**. That $100 you owe
+the bank represents a "bad" thing for you, but a "good" thing for the bank. We might think about
+that same debt differently if we're doing your accounting or the bank's.
+
+These examples also hint at the **different types of accounts**. We probably want to think about a
+debt as having the opposite "sign" as the funds in your your bank account. At the same time, the
+types of these accounts look different depending on whether you are considering them from the
+perspective of you or the bank.
+
+Now, back to our original questions: is the loan balance $100 or -$100 and is the bank account
+balance $200 or -$200? On some level, this feels a bit arbitrary.
+
+Wouldn't it be nice if there were some **commonly agreed-upon standards** so we wouldn't have to
+make such an arbitrary decision? Yes! This is exactly what debits and credits and the financial
+accounting type system provide.
+
+## Types of Accounts
+
+In financial accounting, there are 5 main types of accounts:
+
+- **Assets** - what you own
+- **Liabilities** - what you owe to other people
+- **Equity** - amount invested by the owners or shareholders
+- **Revenue** - money coming in
+- **Expenses** - money going out
```

Comment:
We could improve/correct these definitions a little (and always use the singular, not plural).

For example, Revenue (and the better term to use here would be Income, cf. also "Income Statement") is not in fact money coming in (money coming in could just as well be coming into an Asset account such as a bank account). Rather, an income account is something where the business has received something (and I think perhaps not necessarily money even! but something of value, e.g. an intangible trademark might be given to the company, increasing an asset, and increasing an income) without an obligation arising from that receipt, and where the receipt is probably taxable (cf income tax).

And Equity can also reflect what is owned by the shareholders. For example, not only what they put in, but as the business grows, their ownership of that.

Would be great to lookup definitions for these from the IFRS standards or related discussions (nice rabbit holes in here!). For example: https://www.tandfonline.com/doi/pdf/10.1080/17449480.2010.511892#:~:text=Income%20is%20increases%20in%20economic,from%20equity%20participants.

Reply:
Here's some nice context around the definition of "Income": https://www.investopedia.com/terms/i/income.asp